### PR TITLE
Extract common code for submission commands

### PR DESCRIPTION
**Pull request by barthap**
Chained PR for #56

- Extracted common parts from `IosSubmitter` and `AndroidSubmitter`, created class hierarchy - they now derive from `BaseSubmitter`
- Extracted common methods from `AndroidSubmitCommand` and `IosSubmitCommand` to `commons.ts`
- Implemented "Print Summary" for iOS and extracted common code

### CODE REVIEW

File: packages/eas-cli/src/submissions/ios/IosSubmitter.ts [hunk 1]
```diff
@@ -23,66 +21,29 @@ interface ResolvedSourceOptions {
   appSpecificPassword: string;
 }
 
-class IosSubmitter {
-  constructor(private ctx: IosSubmissionContext, private options: IosSubmissionOptions) {}
+class IosSubmitter extends BaseSubmitter<IosSubmissionContext, IosSubmissionOptions> {
+  protected readonly appStoreName: string = 'Apple TestFlight';
```
Comment (dsokal) [inline packages/eas-cli/src/submissions/ios/IosSubmitter.ts:25]:
```suggestion
  protected readonly appStoreName: string = 'Apple App Store';
```

File: packages/eas-cli/src/submissions/ios/IosSubmitter.ts [hunk 2]
```diff
@@ -23,66 +21,29 @@ interface ResolvedSourceOptions {
...
-    }
-
-    await displayLogs(submission, submissionStatus, verbose);
+    printSummary(
+      submissionConfig,
+      'iOS Submission Summary',
+      SummaryHumanReadableKeys,
+      SummaryHumanReadableValues
+    );
+    // await this.startSubmissionAsync(submissionConfig, this.ctx.commandFlags.verbose);
```
Comment (dsokal) [inline packages/eas-cli/src/submissions/ios/IosSubmitter.ts:44]: Is this commented intentionally?

**Review by dsokal — APPROVED** [review]
👏

File: packages/eas-cli/src/submissions/ios/IosSubmitter.ts (hunk 2 above)
Comment (barthap) [reply packages/eas-cli/src/submissions/ios/IosSubmitter.ts:44]: Yes, as this is only chained PR to a draft, I temporarily commented this out to do some manual tests and don't trigger remote Submission Service unnecessarily. It's gonna be of course uncommented later.

**Pull request merged**